core/domain/auth_jobs_one_off.py

```diff
@@ -39,6 +39,258 @@
+
+        if len(user_id_collisions) > 1:
+            yield '%d UserAuthDetailsModels have auth_id="%s": %s' % (
+                len(user_id_collisions), auth_id, ', '.join(user_id_collisions))
+
+    @classmethod
+    def report_assoc_inconsistencies(
+            cls, auth_id, assoc_by_auth_id=None, assoc_by_user_id=None,
+            firebase_account_disabled=None):
+        """Reports inconsistencies between the given values.
+
+        IMPORTANT: The names of the keyword arguments MUST match the values of
+        their corresponding class keys: ASSOC_BY_AUTH_ID_KEY,
+        ASSOC_BY_USER_ID_KEY, FIREBASE_ACCOUNT_DISABLED_KEY.
+
+        Args:
+            auth_id: str. The auth_id of the user.
+            assoc_by_auth_id: tuple|None. The (user_id, deleted) properties from
+                the UserIdByFirebaseAuthIdModel corresponding to auth_id. NOTE:
+                UserIdByFirebaseAuthIdModel is keyed by auth_id,
+                so this argument will never use the default value of None.
+            assoc_by_user_id: tuple|None. The (user_id, deleted) properties from
+                the UserAuthDetailsModel corresponding to auth_id, or None if
+                one does not exist.
+            firebase_account_disabled: bool|None. Whether the Firebase account
+                corresponding to auth_id is disabled, or None if an account does
+                not exist.
+
+        Yields:
+            str. Debug information about a discovered inconsistency.
+        """
+        if assoc_by_auth_id is None and assoc_by_user_id is None:
+            # User is deleted and will be managed by resolve_user_conflicts().
+            return
+
+        distinct_user_ids = {assoc_by_auth_id[0]}
+        distinct_deleted_bools = {assoc_by_auth_id[1]}
+
+        if assoc_by_user_id is None:
+            yield (
+                'UserIdByFirebaseAuthIdModel(id="%s") does not correspond to a '
+                'unique UserAuthDetailsModel' % auth_id)
+        else:
+            distinct_user_ids.add(assoc_by_user_id[0])
+            distinct_deleted_bools.add(assoc_by_user_id[1])
+
+        if len(distinct_user_ids) == 2:
+            yield (
+                'auth_id="%s" has inconsistent `user_id` assignments: '
+                'UserIdByFirebaseAuthIdModel(user_id="%s") does not match '
+                'UserAuthDetailsModel(id="%s")' % (
+                    auth_id, assoc_by_auth_id[0], assoc_by_user_id[0]))
+
+        if len(distinct_deleted_bools) == 2:
+            yield (
+                'auth_id="%s" has inconsistent `deleted` assignments: '
+                'UserIdByFirebaseAuthIdModel(user_id="%s", deleted=%r) does '
+                'not match UserAuthDetailsModel(id="%s", deleted=%r)' % (
+                    auth_id, assoc_by_auth_id[0], assoc_by_auth_id[1],
+                    assoc_by_user_id[0], assoc_by_user_id[1]))
+
+        elif firebase_account_disabled and not any(distinct_deleted_bools):
```

Comment:
How about the other way around -- the firebase account is not disabled but there's at least one deletion?

Reply:
Elaborated